### PR TITLE
misc cleanup

### PR DESCRIPTION
**Pull request by Alizter**
Here is a bunch of cleanup I did yesterday while looking at the repo. I didn't have enough time to split the commits up, so I'll try to summarise here the main strokes:

- Bumped to dune lang 3.0. Before it was 2.9 which meant that cram tests needed to be manually enabled (the fact that they still partially worked was a bug). In 3.0 cram tests are always enabled so the directory cram tests should work the way you expect. 
- Many of the tests were a bit nonsense with random files included, so I cleaned those up.
- Renabled formatting for dune files, this is something that is useful to have since you don't need to think about it again. Do `dune fmt` every now and then or stick that in CI.
- Generate opam file using Dune package info. All the fields inside the opam file have an equivalent in the dune-project file (which btw you only need one of in this project). The files are generated when you run `dune build` or specifically `dune build @install`.
- Consolidated the scripts used in the test-suite and added them as dependencies of the cram tests in a single dune file rather than having multiple dune files dotted around. Only issue is that the paths to the scripts had to be adjusted.
- Added promoted .exe files to .gitignore

Sorry if I've missed anything.

I tried my best to make sure all the tests looked good, but I think it would be best for you to check it over in case I missed anything.

### CODE REVIEW

File: dune-project [hunk 1]
```diff
@@ -1,3 +1,31 @@
...
+  )
+ (homepage "https://github.com/jmid/mutaml")
+ (documentation "https://github.com/jmid/mutaml")
+ (bug_reports "https://github.com/jmid/mutaml")
+ (authors "Jan Midtgaard <mail@janmidtgaard.dk>")
+ (maintainers "Jan Midtgaard <mail@janmidtgaard.dk>")
+ (tags ("test" "mutation testing"))
+ (depends
+  ppxlib
+  ppx_yosjon_conv
```
Comment (jmid) [inline dune-project:26]: There's a typo here causing the CI to fail immediately:
```diff
- ppx_yosjon_conv
+ ppx_yojson_conv
```

**Review by jmid — COMMENTED** [review]
Thanks, the changes make sense to me :+1: 
Your `dune`-foo is definitely higher than mine! :wink: 

I spent some time identifying lower and upper package bounds to please ocaml-ci in #7:
```
  "ocaml" { >= "4.12.0" }
  "dune" { >= "3.0.0" }
  "ppxlib" { >= "0.26.0" & < "0.28.0" }
  "ppx_yojson_conv" { >= "v0.14.0" }
```
My guess is the bounds were removed by mistake when translating them to `dune-project` and the setup with an auto-generated opam-file.

**Comment by jmid** [comment]
Great, thanks!
I can see ocaml-ci's `(lint-fmt)` check is now failing with `ocamlformat enabled but no .ocamlformat file!`
https://ocaml.ci.dev/github/jmid/mutaml/commit/6741e3335152a4be713780676d7d5c508c7e5ed1/variant/%28lint-fmt%29

If I remember correctly I think I disabled formatting precisely to silence this error and turn that light green as part of #7:
https://github.com/jmid/mutaml/pull/7/commits/11ea1840bf2aeb333cb9be99aaffd13f7842bc0d

Do you know if it is possible to enable auto-formatting of `dune` files and still silence the error?
I don't use `ocamlformat` to format ocaml code, and would prefer to keep it that way for now.

**Comment by Alizter** [comment]
Yes! I'll push such a change. Sorry about that.

**Comment by jmid** [comment]
Looks good, thanks! Just wanted to let the CI finish :+1:

**Pull request merged**